src/Sidekick.Apis.Poe.Trade/Clients/TradeApiHandler.cs

```diff
@@ -0,0 +1,181 @@
+        if (retryResponse.IsSuccessStatusCode)
+        {
+            logger.LogInformation("[PoeTradeHandler] Successfully completed Cloudflare challenge and retried request");
+        }
+        else
+        {
+            logger.LogWarning("[PoeTradeHandler] Request still failed after completing Cloudflare challenge: {StatusCode},\n{RequestHeaders}", retryResponse.StatusCode, request.Headers.ToString());
+        }
+
+        return retryResponse;
+    }
+
+
+    private async Task LogRequest(HttpRequestMessage request, HttpResponseMessage response, CancellationToken cancellationToken = default)
+    {
+        if (response.IsSuccessStatusCode) return;
+
+        logger.LogWarning("[PoeTradeHandler] Failed API call information");
+        logger.LogWarning("[PoeTradeHandler] Uri: {uri}", request.RequestUri);
+
+        if (request.Content != null)
+        {
+            var body = await request.Content.ReadAsStringAsync(cancellationToken);
+            logger.LogWarning("[PoeTradeHandler] Body: {uri}", body);
+        }
+
+        var content = await response.Content.ReadAsStringAsync(cancellationToken);
+        logger.LogWarning("[PoeTradeHandler] Response: {responseCode} {responseMessage}", response.StatusCode, content);
+    }
+
+    private async Task HandleBadRequest(HttpResponseMessage response, CancellationToken cancellationToken)
+    {
+        if (response.StatusCode != HttpStatusCode.BadRequest) return;
+
+        logger.LogWarning("[PoeTradeHandler] BadRequest.");
+        var apiError = await ParseErrorResponse(response, cancellationToken);
+        if (apiError?.Error?.Message?.StartsWith("Query is too complex.") ?? false)
+        {
+            throw new SidekickException("Query is too complex.", "The official trade website has limit on complex queries. Sidekick cannot change this.", "Use the official website to search for your current item.");
+        }
+    }
+
+    private async Task HandleTooManyRequests(HttpResponseMessage response, CancellationToken cancellationToken = default)
+    {
+        if (response.StatusCode != HttpStatusCode.TooManyRequests) return;
+
+        logger.LogWarning("[PoeTradeHandler] TooManyRequests.");
+        var apiError = await ParseErrorResponse(response, cancellationToken);
+        throw new SidekickException("Rate limit exceeded.", "The official trade website has a rate limit to avoid spam. Sidekick cannot change this.", apiError?.Error?.Message ?? string.Empty);
+    }
+
+    private void HandleUnauthorized(HttpResponseMessage response)
+    {
+        if (response.StatusCode != HttpStatusCode.Unauthorized) return;
+
+        logger.LogWarning("[PoeTradeHandler] Unauthorized.");
+        throw new SidekickException("Sidekick failed to communicate with the trade API.", "The trade website requires authentication, which Sidekick does not support currently.", "Try using a different game language and/or force to search using English only in the settings.");
```

Comment:
We should probably adjust the message "The trade website requires authentication" everywhere.